crates/bevy_asset/src/path.rs

```diff
@@ -270,3 +275,131 @@ impl<'de> Visitor<'de> for AssetPathVisitor {
         Ok(AssetPath::from(v))
     }
 }
+
+#[allow(unused_mut)]
+impl GetTypeRegistration for AssetPath<'static> {
```

Comment:
We should have a comment that links to this PR about why we can't do the normal thing here.

Reply:
That exists above!

Reply:
Possible you created this comment (but didn't submit it) before I added that line (which was done after the initial pr).

Reply:
Yep, that was the order of operations here :)